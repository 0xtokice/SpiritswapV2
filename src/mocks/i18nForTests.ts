/* istanbul ignore next */
export const mockUseTranslation = () => ({
  t: key => key,
  i18n: {
    language: undefined,
  },
});
