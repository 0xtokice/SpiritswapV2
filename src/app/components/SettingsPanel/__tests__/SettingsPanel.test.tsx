import { render, fireEvent } from '@testing-library/react';
import { createRenderer } from 'react-test-renderer/shallow';
import {
  SettingsPanel,
  SettingsPanelProps,
} from 'app/components/SettingsPanel';
import { mockUseTranslation } from 'mocks/i18nForTests';
import { ThemeWrapper } from 'app/components/shared/testing/ThemeWrapper';
import { ChakraThemeWrapper } from 'app/components/shared/testing/ChakraTheme';

jest.mock('react-i18next', () => ({
  useTranslation: () => mockUseTranslation(),
}));

const renderer = createRenderer();
const renderComponent = (props: SettingsPanelProps) =>
  render(
    <ChakraThemeWrapper>
      <ThemeWrapper>
        <SettingsPanel {...props} />
      </ThemeWrapper>
    </ChakraThemeWrapper>,
  );

describe('<SettingsPanel />', () => {
  const props = {
    labels: ['0.1%', '0.5%', '1%', 'Auto'],
    selected: 0,
    timer: 20,
    custom: '',
    translationsPath: 'swap.settings',
  };

  it('should render and match the snapshot', () => {
    renderer.render(<SettingsPanel {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should update slippage value on custom input', () => {
    const { getByTestId } = renderComponent(props);
    fireEvent.change(getByTestId('slippage-tolerance-custom'), {
      target: { value: '90%' },
    });
    const element = getByTestId('slippage-tolerance-value');
    expect(element.textContent).toEqual('90%');
  });

  it('should handle back action event', () => {
    const handleClick = jest.fn();
    const { getByTestId } = renderComponent({
      ...props,
      backAction: handleClick,
    });
    fireEvent.click(getByTestId('back-action'));
    expect(handleClick).toHaveBeenCalled();
  });

  it('should reset value', () => {
    const { getByTestId } = renderComponent(props);
    fireEvent.change(getByTestId('slippage-tolerance-custom'), {
      target: { value: '90%' },
    });
    const element = getByTestId('slippage-tolerance-value');
    expect(element.textContent).toEqual('90%');
    fireEvent.click(getByTestId('reset-action'));
    expect(element.textContent).toEqual('0.1%');
  });

  it('should show selected value from <SlippageTolerance /> component', () => {
    const { getByTestId, getByText } = renderComponent(props);
    fireEvent.click(getByText('1%'));
    const element = getByTestId('slippage-tolerance-value');
    expect(element.textContent).toEqual('1%');
  });
});
