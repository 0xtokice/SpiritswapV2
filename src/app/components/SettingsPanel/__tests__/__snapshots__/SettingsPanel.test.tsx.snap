// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsPanel /> should render and match the snapshot 1`] = `
<Styled(Panel)>
  <Styled(div)>
    <Styled(div)>
      <Button
        data-testid="back-action"
        flat={true}
        variant="inverted"
      >
        <Styled(Icon)
          icon={<ForwardRef(SvgArrowRight) />}
          size="normal"
        />
      </Button>
      swap.settings.title
    </Styled(div)>
    <Button
      data-testid="reset-action"
      flat={true}
      onClick={[Function]}
      variant="inverted"
    >
      swap.settings.reset
    </Button>
  </Styled(div)>
  <Styled(div)>
    <Styled(div)>
      <Styled(div)>
        <Styled(Paragraph)>
          swap.settings.slippageToleranceLabel
        </Styled(Paragraph)>
        <QuestionHelper
          text="swap.settings.slippageExplanation"
          title="swap.settings.slippageToleranceLabel"
        />
      </Styled(div)>
      <Styled(div)
        data-testid="slippage-tolerance-value"
      >
        0.1%
      </Styled(div)>
    </Styled(div)>
    <Styled(div)>
      <SlippageTolerance
        customValue="0%"
        labels={
          Array [
            "0.1%",
            "0.5%",
            "1%",
            "Auto",
          ]
        }
        onChange={[Function]}
        selected={0}
      />
    </Styled(div)>
    <Styled(div)>
      <Styled(div)>
        <Styled(Paragraph)>
          swap.settings.transactionDeadlineLabel
        </Styled(Paragraph)>
        <QuestionHelper
          text="swap.settings.transactionDeadlineExplanation"
          title="swap.settings.transactionDeadlineLabel"
        />
      </Styled(div)>
      <Styled(div)>
        <Styled(Icon)
          icon={<ForwardRef(SvgTimer) />}
          size="normal"
        />
        <Styled(div)>
          20
        </Styled(div)>
      </Styled(div)>
    </Styled(div)>
    <Styled(div)>
      <Styled(div)>
        <Styled(Paragraph)>
          swap.settings.chartModeLabel
        </Styled(Paragraph)>
        <QuestionHelper
          text="swap.settings.chartExplanation"
          title="swap.settings.chartModeLabel"
        />
      </Styled(div)>
      <Styled(div)>
        <Switch
          label="OFF"
        />
      </Styled(div)>
    </Styled(div)>
    <Styled(div)>
      <Styled(div)>
        <Styled(Paragraph)>
          swap.settings.tradeModeLabel
        </Styled(Paragraph)>
        <QuestionHelper
          text="swap.settings.tradeModeExplanation"
          title="swap.settings.tradeModeLabel"
        />
      </Styled(div)>
      <Styled(div)>
        <Switch
          label="Expert"
        />
      </Styled(div)>
    </Styled(div)>
  </Styled(div)>
</Styled(Panel)>
`;
